utils/clusterutils.go

```diff
@@ -42,11 +43,32 @@ func watchDaemonset(clientset *kubernetes.Clientset) watch.Interface {
 	return watch
 }
 
-func getDaemonset() *appsv1.DaemonSet {
+func getImagePullerDeployment(clientset *kubernetes.Clientset) *appsv1.Deployment {
+	cfg := cfg.GetConfig()
+	deployment, err := clientset.AppsV1().Deployments(cfg.Namespace).Get(os.Getenv("DEPLOYMENT_NAME"), metav1.GetOptions{})
```

Comment:
You may want to check that `os.Getenv("DEPLOYMENT_NAME")` is not `""` for a slightly clearer error message